package com.ecommerce.repository;

import com.ecommerce.model.Cart;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface CartRepository extends MongoRepository<Cart, String> {
    
    Optional<Cart> findByUserIdAndStatus(String userId, Cart.CartStatus status);
    
    List<Cart> findByStatus(Cart.CartStatus status);
    
    List<Cart> findByUserId(String userId);
}

